src/main/java/com/api/service/NetworkingService.java

```diff
@@ -0,0 +1,50 @@
+package com.api.service;
+
+import com.api.entity.SX_Rocket;
+import com.api.utility.UriResolver;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+
+@Service
+@PropertySource("classpath:application.properties")
+public class NetworkingService {
+
+    @Value("${rocket.api.base-url}")
+    private String getBaseUrl;
+
+    @Value("${rocket.api.default-limit}")
+    private int getDefaultLimit;
+
+    @Value("${rocket.api.default-offset}")
+    private int getDefaultOffset;
+
+    private final WebClient webClient;
+
+    public NetworkingService(WebClient.Builder webClientBuilder) {
+        this.webClient = webClientBuilder.baseUrl(getBaseUrl).build();
```

Comment:
move the base init into the config, as the base will be always the same in any service